beacon-chain/db/kv/state.go

```diff
@@ -308,20 +308,21 @@ func slotByBlockRoot(ctx context.Context, tx *bolt.Tx, blockRoot []byte) (uint64
 		return stateSummary.Slot, nil
 	}
 
-	bkt := tx.Bucket(stateBucket)
+	bkt := tx.Bucket(blocksBucket)
```

Comment:
We actually started out by using block. But one very good feedback from @prestonvanloon: what if the block is missing then the node will never be able to delete the state

I know the state is slow, I think the compromise (later) is to add `root->slot` storage

Reply:
This is an issue during sync, unmarshalling a state just to delete it is quite inefficient. If a block is missing, then how would we save its resulting state ? 
i could add a fallback to retrieve the state in the event the block is missing.

Reply:
This is just preparing for the worst case (ie, user saves the state then mistakenly deletes the block.. etc)

fallback sounds good

Reply:
ok done, added in a fallback